semgrep_output_v1.atd

```diff
@@ -919,6 +929,55 @@ type scan_request = {
     ?scan_metadata: scan_metadata option;
 }
 
+(* Response by the backend to the CLI to the POST /scans *)
+(* TODO
+type scan_response = {
+    scan: scan_response_bis;
+}
+type scan_response_bis = {
+    (* the scan id *)
+    id: string;
+    (* since 1.37.0 *)
+    ?enabled_products: product list;
+}
+*)
+
+(* TODO? seems used only for dry_run and passed urlencoded to /scans/config?xxx
+type scan_config_request = {
+    dry_run: bool;
+    (* project_metadata.repository *)
+    repo_name: string;
+    (* TODO: deprecate, prefer scan_request.scan_metadata.requested_products *)
+    sca: bool;
+    full_scan: bool;
+    semgrep_version : version;
+}
+ *)
+
+(* Response by the backend to the CLI to the POST /scans/<scan_id>/config *)
+type scan_config = {
+    deployment_id: int;
+    (* the org name *)
+    deployment_name: string;
+    (* ex: "audit", "comment", "block" TODO use enum? *)
+    policy_names: string list;
+    (* rules raw content in JSON format *)
+    rule_config: string;
```

Comment:
This isn't always a string - I started migrating this to json for semgrep scan a while back and put the brakes on after that. Will dig into this today.

Reply:
We should not use different types ... In any case we could add another field with rule_as_json: